syntax = "proto3";

package bluetooth.facade;

message BluetoothAddress {
  bytes address = 1;
}

enum BluetoothAddressTypeEnum {
  PUBLIC_DEVICE_ADDRESS = 0x0;
  RANDOM_DEVICE_ADDRESS = 0x1;
  PUBLIC_IDENTITY_ADDRESS = 0x2;
  RANDOM_IDENTITY_ADDRESS = 0x3;
}

message BluetoothAddressWithType {
  BluetoothAddress address = 1;
  BluetoothAddressTypeEnum type = 2;
}

enum BluetoothPeerAddressTypeEnum {
  PUBLIC_DEVICE_OR_IDENTITY_ADDRESS = 0x0;
  RANDOM_DEVICE_OR_IDENTITY_ADDRESS = 0x1;
}
